src/lib/registration/ui/migration_selection_dialog.rb

```diff
@@ -181,7 +182,7 @@ def migration_details(idx)
 
         details = sorted_migrations[idx].map do |product|
           installed = installed_products.find do |installed_product|
-            installed_product["name"] == product.identifier
+            same_products?(installed_product, product)
```

Comment:
NP: in object world this should be method of product like e.g. `=~`

Reply:
In this cases the objects are of different classes, that makes it a bit more complicated... But yes you are right.